File: JavaScript/uri-1022.js
// URI - 1022: https://www.urionlinejudge.com.br/judge/pt/problems/view/1022

const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

const calculator = () => {
    const mathematicalExpression = lines.shift().split(' ')
    const numerator1 = parseInt(mathematicalExpression[0])
    const denominator1 = parseInt(mathematicalExpression[2])
    const operator = mathematicalExpression[3]
    const numerator2 = parseInt(mathematicalExpression[4])
    const denominator2 = parseInt(mathematicalExpression[6])

    let GCDresult
    let operationPart1
    let operationPart2

    if (operator == '+') {
        operationPart1 = parseInt(numerator1 * denominator2 + numerator2 * denominator1)
        operationPart2 = parseInt(denominator1 * denominator2)
    } else if (operator == '-') {
        operationPart1 = parseInt(numerator1 * denominator2 - numerator2 * denominator1)
        operationPart2 = parseInt(denominator1 * denominator2)
    } else if (operator == '*') {
        operationPart1 = parseInt(numerator1 * numerator2)
        operationPart2 = parseInt(denominator1 * denominator2)
    } else if (operator == '/') {
        operationPart1 = parseInt(numerator1 * denominator2)
        operationPart2 = parseInt(numerator2 * denominator1)
    } else {
        console.log('Invalid operator!')
    }
    
    GCDresult = parseInt(GCD(operationPart1, operationPart2))
    let simplificationPart1 = parseInt(operationPart1 / GCDresult)
    let simplificationPart2 = parseInt(operationPart2 / GCDresult)

    if (simplificationPart1 > 0 && simplificationPart2 > 0) {
        console.log(`${operationPart1}/${operationPart2} = ${simplificationPart1}/${simplificationPart2}`)
    } else {
        if (simplificationPart2 < 0) {
            simplificationPart1 = -simplificationPart1
            simplificationPart2 = -simplificationPart2
        }
        console.log(`${operationPart1}/${operationPart2} = ${simplificationPart1}/${simplificationPart2}`)
    }
}

const GCD = (part1, part2) => {
    let divider, dividend, rest

    if (part1 == 0) {
        return 1
    }

    if (part2 > part1) {
        dividend = part2
        divider = part1
    } else {
        divider = part2
        dividend = part1
    }

    while (dividend % divider != 0) {
        rest = dividend % divider
        dividend = divider
        divider = rest
    }

    return divider
}

const numberOfCalculations = parseInt(lines.shift())

for (let i = 0; i < numberOfCalculations; i++) {
    calculator()
}

File: Ruby/uri-1022.rb
#URI - 1022: https://www.urionlinejudge.com.br/judge/pt/problems/view/1022

def calculate_mathematical_expression(mathematical_expression)
    values = [mathematical_expression[0].to_i, mathematical_expression[2].to_i,
              mathematical_expression[4].to_i, mathematical_expression[6].to_i]
    operation = mathematical_expression[3]
    case operation
    when "+"
        result = sum(values)
    when "-"
        result = subtract(values)
    when "*"
        result = multiply(values)
    when "/"
        result = divide(values)
    else
        puts "Invalid operator"
    end
    return result
end

def sum(values)
    return [(values[0] * values[3] + values[2] * values[1]), (values[1] * values[3])]
end

def subtract(values)
    return [(values[0] * values[3] - values[2] * values[1]), (values[1] * values[3])]
end

def multiply(values)
    return [(values[0] * values[2]), (values[1] * values[3])]
end

def divide(values)
    return [(values[0] * values[3]), (values[2] * values[1])]
end

def simplify_result(result)
    return Rational(result[0], result[1])
end

def show_results(result, simplified_result)
    puts "#{result[0]}/#{result[1]} = #{simplified_result}"
end

Number_of_test_cases = gets.to_i
Number_of_test_cases.times do
    input = gets.split(' ')
    result = calculate_mathematical_expression(input)
    simplified_result = simplify_result(result)
    show_results(result, simplified_result)
end